newIDE/app/src/InstancesEditor/InstancesRotator.js

```diff
@@ -0,0 +1,44 @@
+export default class InstancesRotator {
+  constructor({ options }) {
+    this.options = options;
+    this.instanceAngles = {};
+    this.totalDeltaX = 0;
+    this.totalDeltaY = 0;
+  }
+
+  setOptions(options) {
+    this.options = options;
+  }
+
+  _getAngleDelta(proportional, initialAngle) {
+    let deg = Math.atan2(this.totalDeltaY, this.totalDeltaX) * 180 / Math.PI;
```

Comment:
Also concerning the method name, I think you can simply name it `_getNewAngle`. Delta is a term and symbol often used (notably in mathematics) to say that a quantity is a difference between two other quantities (in this class for example, `totalDeltaX` and `totalDeltaY` means the "difference" of position on X and Y axis (of the cursor)).

"New angle" is better suited as what we're getting is an angle and not a difference (and you properly named newAngle below by the way ;)).

```
_getNewAngle(proportional: boolean, initialAngle: number) {
  const angle = Math.atan2(this.totalDeltaY, this.totalDeltaX) * 180 / Math.PI + initialAngle;
  return proportional ? Math.round(angle / 45) * 45 : angle;
}
```

Reply:
good point, sorry I forgot to add the flow check :)
I was too fixated on the idea that the angle that GD uses is in degrees,not in radiants and that I am using the mouse cursor movement to get it, not 4 points in space (x1,x2,y1,y2). 

The formula really is
`Math.atan2(y2-y1, x2-x1)`
However since delta is relative to the button click start, that means that y1 and x1 will always be 0 anyway.
The other part just coverts it to degrees

The way the angle is computed is in using the delta motion of the mouse, this could be changed if it causes any issues :)